### ISSUE TITLE
[BUG] 같은 코드와 hyper parameter임에도 다르게 성능이 측정되는 현상

### ISSUE THREAD

**Issue by GY-Jeong** [bcaitech1/p4-dkt-lastffang#100]
# 원인 

`train.py`에서 나누어진 `train_data`와 `valid_data`가

`dataloader.py`의 `get_loaders`를 거치며 `DKTDataset`으로 바뀌고, `DataLoader`를 통해 `train_loader`, `valid_loader`로 바뀌는데,

이 때 `torch.utils.data.DataLoader( , , , shuffle=True, ,)`에서 발생

# 검증 

저의 서버 + 건희님 서버 + 저의 로컬 + 연구실 컴퓨터로 실험해봤을 때, train_data, valid_data와 train_loader에서 하나의 batch를 출력하여 비교

<저의 서버>

```
Start Training: Epoch 1
[tensor([[0., 1., 1.,  ..., 1., 1., 1.],
        [1., 1., 0.,  ..., 1., 0., 0.],
        [0., 1., 1.,  ..., 1., 1., 1.],
        ...,
        [1., 1., 0.,  ..., 0., 0., 0.],
        [1., 1., 1.,  ..., 1., 0., 0.],
        [1., 1., 1.,  ..., 0., 1., 0.]]), tensor([[ 718.,  718.,  718.,  ...,  738.,  738.,  738.],
        [ 191.,  191.,  191.,  ..., 1460., 1460., 1460.],
        [ 672.,  672.,  676.,  ..., 1402., 1402., 1402.],
```

<건희님 서버>

```
Start Training: Epoch 1
[tensor([[1., 1., 0.,  ..., 0., 0., 1.],
        [1., 1., 1.,  ..., 0., 1., 0.],
        [1., 1., 1.,  ..., 1., 1., 1.],
        ...,
        [1., 1., 1.,  ..., 0., 1., 1.],
        [1., 1., 1.,  ..., 1., 1., 1.],
        [0., 0., 0.,  ..., 1., 1., 1.]]), tensor([[ 506.,  506.,  506.,  ...,  508.,  508.,  508.],
        [1100., 1100., 1100.,  ..., 1104., 1104., 1104.],
        [ 749.,  749.,  749.,  ...,  324.,  324.,  324.],
```

`train_data`와 `valid_data`가 동일함에도 `train_lodaer`로부터 읽어오는 첫번째 batch는 다른 것을 확인할 수 있음

# 해결 방안
당장 적용할 수 있는 것은 `torch.utils.data.DataLoader`의 shuffle 옵션을 False로 만드는 것

```
def get_loaders(args, train, valid):
    pin_memory = True
    train_loader, valid_loader = None, None

    if train is not None:
        trainset = DKTDataset(train, args)
        train_loader = torch.utils.data.DataLoader(trainset, num_workers=args.num_workers, shuffle=False,
                                                   batch_size=args.batch_size, pin_memory=pin_memory,
                                                   collate_fn=collate)
```

랜덤하게 섞고, 각 서버가 동일하게 가져올 수 있는 방안은 따로 찾아봐야 함
